crates/sui-indexer/src/indexer.rs

```diff
@@ -86,17 +88,23 @@ impl Indexer {
             .unwrap_or(CHECKPOINT_PROCESSING_BATCH_DATA_LIMIT.to_string())
             .parse::<usize>()
             .unwrap();
+        let extra_reader_options = ReaderOptions {
+            batch_size: download_queue_size,
+            timeout_secs: ingestion_reader_timeout_secs,
+            data_limit,
+            ..Default::default()
+        };
 
-        let rest_client = sui_rest_api::Client::new(format!("{}/rest", config.rpc_client_url));
-
-        let objects_snapshot_processor = ObjectsSnapshotProcessor::new_with_config(
-            rest_client.clone(),
+        // Start objects snapshot processor, which is a separate pipeline with its ingestion pipeline.
+        start_objects_snapshot_processor::<S, T>(
```

Comment:
getting ahead of myself but it looks like it'd be possible to not even need to start an `objects_snapshot_processor` depending on the indexer configs

Reply:
I was thinking about this possibility too. We don't need this right now since our indexer writer instance will be serving both graphql and json servers. But if we are serving a lot of folks who don't need objects snapshot at all (i.e. because they only run jsonrpc), then a config is certainly a good idea.